coffeecard/CoffeeCard.Library/Services/v2/ProductService.cs

```diff
@@ -81,41 +81,43 @@ public async Task<ChangedProductResponse> AddProduct(AddProductRequest newProduc
                 Visible = newProduct.Visible
             };
 
-            _context.Products.Add(product);
-            await _context.SaveChangesAsync();
-
-            var productUserGroups = newProduct.AllowedUserGroups.Select(userGroup => new ProductUserGroup
+            product.ProductUserGroup = newProduct.AllowedUserGroups.Select(userGroup => new ProductUserGroup
             {
                 ProductId = product.Id,
                 UserGroup = userGroup
             }).ToList();
 
-            _context.ProductUserGroups.AddRange(productUserGroups);
-
-
+            _context.Products.Add(product);
```

Comment:
I am uncertain what you mean with "re-add the Products":
This product is not added to the context yet. It is created in line 74, added user groups to the product on line 84 and then added to the context on line 90 before being saved.

My new test, `AddProduct adds only selected user groups`, fails with error message `CoffeeCard.Common.Errors.EntityNotFoundException : No product was found by Product Id: 1 ` if I remove this line where I add it to the context. Could it be that since the changes now also involves the `AddProduct` method that you confused this with the `UpdateProduct` method?

Reply:
You are right. I messed up the lines. Looks good now! 😊